packages/labs/analyzer/src/lib/analyzer.ts

```diff
@@ -21,9 +28,8 @@ export {PackageJson};
 export class Analyzer {
   readonly packageRoot: AbsolutePath;
   readonly commandLine: ts.ParsedCommandLine;
-  readonly program: ts.Program;
-  readonly checker: ts.TypeChecker;
   readonly packageJson: PackageJson;
+  readonly programContext: ProgramContext;
```

Comment:
It's mostly FUD about avoiding circular imports in our internal toolchain (`analyzer.ts` would `import {getEvents} from './events.js'`, and `events.ts` would `import {Analyzer} from './analyzer.js'` (to type the `analyzer` argument to `getEvents`). Checking with @rictic whether this is actually an issue to avoid in g3 or whether I just dreamed it. Maybe it's not an issue if the circular dep is just the type? Or if they're all in the same ts_library? 

Otherwise yeah, this could all just be in the analyzer. FWIW I also want to factor `Module` and the various `Declaration` features out to their own module, started that here (will wait till this one lands): https://github.com/lit/lit/pull/2994

Reply:
Peter thinks circular refs are likely going to be stressful. For now, added a TODO and leaving this as-is. Totally up for refactoring it later.